File: Day008_2025/Q008_Leetcode.java
class Q008_Leetcode {
    public int countPrefixSuffixPairs(String[] words) {
        int count = 0;

        for (int i = 0; i < words.length; ++i) {
            for (int j = i + 1; j < words.length; ++j) {
                String str1 = words[i];
                String str2 = words[j];

                // Check if str1 is prefix and suffix of str2
                if (str1.length() > str2.length()) 
                    continue;

                if (str2.startsWith(str1) && str2.endsWith(str1)) {
                    count++;
                }
            }
        }

        return count;
    }

    public static void main(String[] args) {
        String[] words = {"abc", "abcabc", "bc", "abcabcabc"};
        Q008_Leetcode obj = new Q008_Leetcode();
        System.out.println(obj.countPrefixSuffixPairs(words)); // Example usage
    }
}

File: Day008_2025/Q008_Leetcode.py
from typing import List

class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        count = 0

        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                str1 = words[i]
                str2 = words[j]

                # Check if str1 is prefix and suffix of str2
                if len(str1) > len(str2):
                    continue

                if str2.startswith(str1) and str2.endswith(str1):
                    count += 1

        return count

# Example usage
if __name__ == "__main__":
    words = ["abc", "abcabc", "bc", "abcabcabc"]
    obj = Solution()
    print(obj.countPrefixSuffixPairs(words))  # Output example
